src/Module/Profile/Photos/Index.php

```diff
@@ -52,16 +61,268 @@ class Index extends \Friendica\Module\BaseProfile
 	private $app;
 	/** @var Database */
 	private $database;
+	/** @var SystemMessages */
+	private $systemMessages;
+	/** @var ACLFormatter */
+	private $aclFormatter;
 
-	public function __construct(Database $database, App $app, IManageConfigValues $config, App\Page $page, IHandleUserSessions $session, L10n $l10n, App\BaseURL $baseUrl, App\Arguments $args, LoggerInterface $logger, Profiler $profiler, Response $response, array $server, array $parameters = [])
+	public function __construct(ACLFormatter $aclFormatter, SystemMessages $systemMessages, Database $database, App $app, IManageConfigValues $config, App\Page $page, IHandleUserSessions $session, L10n $l10n, App\BaseURL $baseUrl, App\Arguments $args, LoggerInterface $logger, Profiler $profiler, Response $response, array $server, array $parameters = [])
 	{
 		parent::__construct($l10n, $baseUrl, $args, $logger, $profiler, $response, $server, $parameters);
 
-		$this->session  = $session;
-		$this->page     = $page;
-		$this->config   = $config;
-		$this->app      = $app;
-		$this->database = $database;
+		$this->session        = $session;
+		$this->page           = $page;
+		$this->config         = $config;
+		$this->app            = $app;
+		$this->database       = $database;
+		$this->systemMessages = $systemMessages;
+		$this->aclFormatter   = $aclFormatter;
+	}
+
+	protected function post(array $request = [])
+	{
+		$user = User::getByNickname($this->parameters['nickname'] ?? '');
+		if (!$user) {
+			throw new HTTPException\NotFoundException($this->t('User not found.'));
+		}
+
+		$page_owner_uid = intval($user['uid']);
+		$visitor   = $this->session->getRemoteContactID($page_owner_uid);
```

Comment:
In the original code `$visitor` is set to `0` for local users. I would prefer keeping it that way to avoid any side effects.

Reply:
This is already the case in `getRemoteContactID`.

Reply:
I haven't checked, but I wouldn't rely on it in any way 😬